docs/use_cases/fine_grained_authorization/dynamic.md

```diff
@@ -0,0 +1,416 @@
+a very small subset of these relationships will typically be accessed at runtime. We therefore want to optimize our
+computation to only derive the subset of relationships that can be accessed by authorization checks performed by the system.
+For example, only users who are logged into the system can issue authorization requests; hence we only need to consider
+currently active users in the computation.  When a new user logs in, their permissions can be incrementally computed on the fly.
+
+Similarly, not all resources in the system can be addressable at any given time, for example it's possible that users can
+only access files when they are browsing their parent folder.  In this case, we should only track permissions for currently
+active resources.  Let's implement this last optimization as an example.  Replace the `create table objects(..);` declaration
+with the following fragment:
+
+```sql
+-- All objects in the system, including active and currently inactive objects.
+create table all_objects (
+    id id_t not null primary key,
+    properties properties_t
+);
+
+-- Currently active objects.
+--
+-- This table is a subset of object ids, including only those objects for which authorization rules need to be evaluated.
+-- The definition of an "active object" varies depending on the application and may include, for example, folders or wiki pages
+-- currently accessed or open by at least one user.
+create table active_objects(
+    object_id id_t not null
+);
+
+-- Relevant objects are all active objects plus all objects from which
+-- an active object can be reached by following object graph edges.
+declare recursive view relevant_objects(
+    object_id id_t not null
+);
+
+create view relevant_objects as
+select * from active_objects
+union all
+select relationships.subject_id
+    from relevant_objects join relationships on relevant_objects.object_id = relationships.resource_id;
+
+-- Objects whose id's are in `relevant_objects`.
+create materialized view objects as
+select all_objects.*
+    from all_objects join relevant_objects on all_objects.id = relevant_objects.object_id;
+```
+
+The new program will only compute derived relationships over objects in `relevant_objects` only.
+
+## Takeaways
+
+Whether you are adding a flexible authorization layer to your application or building a general-purpose policy
+framework, Feldera offers a high-performance off-the-shelf compute engine for the job. Two features of Feldera
+make it ideal for this purpose:
+
+1. **Incremental query evaluation** enables Feldera to handle changes to the object graph or the FGA model
+ in real-time by avoiding complete recomputation.
+2. **Support for mutually recursive views** allows naturally capturing iterative graph traversal with recursive queries.
+
+With these capabilities, Feldera stands out as a powerful solution for advanced authorization and policy management tasks.
```

Comment:
how about a solution where you only compute part of the graph when specifying some endpoints (using another table)?